event-sourcing/event-source/src/main/java/uk/gov/justice/services/eventsourcing/source/core/EventSourceProducer.java

```diff
@@ -61,15 +55,20 @@ public EventSource eventSource() {
     public EventSource eventSource(final InjectionPoint injectionPoint) {
 
         final String eventSourceName = qualifierAnnotationExtractor.getFrom(injectionPoint, EventSourceName.class).value();
-
-        final Optional<EventSourceDefinition> eventSourceDefinition = eventSourceDefinitionRegistry.getEventSourceDefinitionFor(eventSourceName);
-        return eventSourceDefinition
-                .map(this::createEventSourceFrom)
-                .orElseThrow(() -> new CreationException(format("Failed to find EventSource named '%s' in event-sources.yaml", eventSourceName)));
+        final EventSourceDefinition eventSourceDefinition;
+
+        if (eventSourceName.isEmpty()) {
```

Comment:
Do we want to support empty event source names? What if we make the name mandatory?